File: com.issue.tracker.api/src/main/java/com/issue/tracker/api/logger/Log.java
package com.issue.tracker.api.logger;

import java.io.Serializable;
import java.time.LocalDateTime;

public abstract class Log implements LogOperations, Serializable {
    protected LogType logType;
    protected String message;
    protected Class<?> invokerClass;
    protected LocalDateTime timestamp;
    protected String stackTrace;
    protected String reason;

    public Log(Class<?> invokerClass, LogType logType, String message) {
        this.invokerClass = invokerClass;
        this.logType = logType;
        this.message = message;
        this.timestamp = LocalDateTime.now();
    }

    public void setLogType(LogType logType) {
        this.logType = logType;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public void setInvokerClass(Class<?> invokerClass) {
        this.invokerClass = invokerClass;
    }

    public void setStackTrace(String stackTrace) {
        this.stackTrace = stackTrace;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }
}

File: com.issue.tracker.api/src/main/kotlin/com/issue/tracker/api/logger/Log.kt
package com.issue.tracker.api.logger

import kotlinx.datetime.Clock
import kotlinx.datetime.LocalDateTime
import kotlinx.datetime.TimeZone
import kotlinx.datetime.toLocalDateTime
import kotlin.reflect.KClass

abstract class Log(
    invokerClass: KClass<*>,
    logType: LogType,
    message: String,
) : LogOperations {
    var logType: LogType = logType
        protected set
    var message: String = message
        protected set

    var invokerClass: KClass<*> = invokerClass
        protected set

    val timestamp: LocalDateTime = Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())

    var stackTrace: String? = null

    var reason: String? = null
}